service/history/queues/executable.go

```diff
@@ -341,6 +348,13 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 		e.logger.Error("Drop task due to serialization error", tag.Error(err))
 		return nil
 	}
+	if common.IsInternalError(err) {
+		e.logger.Error("Encountered internal error processing tasks", tag.Error(err))
```

Comment:
should not this log.Error be inside the `if e.dropInternalErrors()`? if you are not returning you log the error anyway in line 361.

Reply:
Good catch! Looks like we missed this in my patch to v1.23 as well. I'll follow up with 1.23 to fix that when I next have a chance

Reply:
I think idea here is that even when `dropInternalErrors` is `false`, we still log and emit metric for the internal errors so that we can be confident that they are safe to drop. and then enable the flag. It's a shadow mode basically.

Having this log outside `dropInternalErrors` makes it easier to filter logs and only look at internal error logs. With only the log on L361, I don't think there's a good way to audit just the internal errors. The Error() method of an InternalError doesn't say it's an internal error 🤦.

Reply:
We could add a tags.ErrorType tag to the main log message here. That'd give us a single log that handles both cases.

I'm going to do so in a separate PR and port over the code from https://github.com/temporalio/temporal/pull/5234/files#diff-3045f1928c46472037eb67e844c90f8745e01cd321e058ba212e3ea1a6b5147fR51 to unwrap things like `fmt.wrapErr`

Reply:
Put up a pr at https://github.com/temporalio/temporal/pull/5400